### PR TITLE
Reenable tasks without side-effects, that only run code

### LINKED ISSUE #7848: Mix run can't execute multiple scripts
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 

```
Erlang/OTP 20 [erts-9.3.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.6.5 (compiled with OTP 20)
```

* Operating system: MacOS

### Current behavior

If I have a mix alias like this:

```
"my.alias": ["run script1.exs", "run script2.exs"]
```

only the first script will be run.

### Expected behavior

I would expect both scripts to run.

As mentioned in several places [[1]] [[2]] [[3]], mix tasks can only be run once. I'm not sure why the limitation exists, but in the context of `run script.exs` it doesn't make sense to me.

[1]: https://stackoverflow.com/questions/48291606/how-to-alias-run-twice-in-mix-exs
[2]: https://stackoverflow.com/questions/36846041/elixir-mix-aliases-two-run-tasks-in-one-alias
[3]: https://github.com/elixir-lang/elixir/issues/6488

While it makes sense that we wouldn't want the exact same mix task to run twice, it seems like it should be conditional on the task __and arguments__ rather than just the task.

### PR DESCRIPTION
**Pull request by josevalim**
Closes #7848.

### CODE REVIEW

File: lib/mix/lib/mix/tasks/run.ex [hunk 1]
```diff
@@ -40,11 +40,14 @@ defmodule Mix.Tasks.Run do
...
 
+  This task is automatically reenabled, so it can be called multiple times
+  with different arguments.
+
   ## Command-line options
 
     * `--config`, `-c`  - loads the given configuration file
     * `--eval`, `-e` - evaluates the given code
-    * `--require`, `-r` - requires pattern before running the command
+    * `--require`, `-r` - executes the given pattern/file
```
Comment (fertapric) [inline lib/mix/lib/mix/tasks/run.ex]: I wonder if this change is consistent with the description of the next flag:

> `parallel`, `-p` - makes all requires parallel

Maybe “requires (and executes) the given pattern/file”?

Comment (josevalim) [reply lib/mix/lib/mix/tasks/run.ex]: I think so, since the word "requires" is talking about the `--require` flag. Thoughts?

**Comment by michalmuskala** [comment]
I think this is a sensible change, I don't see any downsides to this.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**